server/src/main/java/org/opensearch/common/blobstore/AsyncMultiStreamEncryptedBlobContainer.java

```diff
@@ -108,4 +106,69 @@ public InputStreamContainer provideStream(int partNumber) throws IOException {
         }
 
     }
+
+    static class DecryptedReadContext<T, U> extends ReadContext {
+
+        private final U cryptoContext;
+        private final CryptoHandler<T, U> cryptoHandler;
+        private final long fileSize;
+
+        public DecryptedReadContext(ReadContext readContext, CryptoHandler<T, U> cryptoHandler) {
+            super(readContext);
+            this.cryptoHandler = cryptoHandler;
+            try {
+                this.cryptoContext = this.cryptoHandler.loadEncryptionMetadata(null);
```

Comment:
Q: How do we load the metadata here?

Reply:
You need to pass a content supplier here. You can look at EncryptedBlobContainer#readBlob . The supplier will invoke a ranged read of blobcontainer.